notebooks/sycamore-tutorial-intermediate-etl.ipynb

```diff
@@ -0,0 +1,410 @@
+    "We make a remote call to APS for each document. APS will take a few minutes to partition these documents, as there are a few hundred pages across the PDF doucments. \n",
+    "\n",
+    "Notice that we include a `materialize` operation and force execution with `execute`. `materialize` writes the DocSet out to disk, and it is configured to be used as a checkpoint in the pipeline. Subsequent operations will read from this materialized DocSet. In this notebook, we will use commands like `show_pages` to visualize the DocSet for educational purposes (a production pipeline wouldn't do this), and these commands force recomputation of the DocSet. If present, Sycamore will use the materialized DocSet instead of recomputing it (and avoiding running the APS step again)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can visualize the pages of the PDF using the `show_pages` function, so you can see how APS chunked each page into elements.\n",
+    "\n",
+    "Note that this requires the `poppler` library to be installed on your system. You can install it on a Mac with `brew install poppler`, or in Debian with `apt-get install poppler-utils`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.utils.pdf_utils import show_pages\n",
+    "\n",
+    "show_pages(partitioned_docset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Cleaning data\n",
+    "\n",
+    "We often find that there's a little excess whitespace in the textual representation of documents, which can be bad for our subsequent LLM transforms and embeddings. So, we apply a regex on the text representation of each element to coalesce it with the `regex_replace` transform."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.transforms.regex_replace import COALESCE_WHITESPACE\n",
+    "\n",
+    "regex_docset = partitioned_docset.regex_replace(COALESCE_WHITESPACE)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Metadata extraction\n",
+    "\n",
+    "For many use cases, extracting additional metadata from each document (and storing it as document metadata/properties) is important. To do this, we can apply the `extract_properties` transform, which sends the documents to an LLM to extract properties determined by the schema we provide in the prompt. (The LLM could also create the schema with the `extract_batch_schema` transform) \n",
+    "\n",
+    "We create the schema using the `with_property` transform and attach it to each document as a property. Then, we `extract_properties` using OpenAI's GPT. You'll need to set your OpenAI key in your env or in the Op The `num_of_elements` parameter determines how many elements from each document will be sent to the LLM for extraction."
```

Comment:
Also worth mentioning you can get these properties using the `extract_batch_schema` transform. They look very specific otherwise